packages/mui-joy/src/Autocomplete/AutocompleteProps.ts

```diff
@@ -95,15 +82,17 @@ interface ComponentsProps {
       IconButtonOwnerState,
       'color' | 'variant' | 'size'
     >,
-    AutocompleteOwnerState<any, any, any, any> & IconButtonOwnerState
+    Omit<AutocompleteOwnerState<any, any, any, any>, 'color' | 'variant' | 'size'> &
+      Pick<IconButtonOwnerState, 'color' | 'variant' | 'size'>
```

Comment:
Pick only the styling props from the IconButton.

Reply:
Why is IconButtonOwnerState needed here? ownerState is supposed to be the state of the owner - that is Autocomplete in this case.

Reply:
Right, move them to props.